Signal/src/call/UserInterface/CallUIAdapter.swift

```diff
@@ -39,15 +39,22 @@ extension CallUIAdaptee {
         let callViewController = CallViewController(call: call)
         callViewController.modalTransitionStyle = .crossDissolve
 
-        guard let presentingViewController = UIApplication.shared.frontmostViewControllerIgnoringAlerts else {
+        guard let mainWindow = CurrentAppContext().mainWindow else {
+            owsFail("in \(#function) mainWindow unexpectedly nil")
+            return
+        }
+        guard let presentingViewController = mainWindow.rootViewController else {
```

Comment:
Per our discussion, this should be fine - and make us robust to our presenting VC being dismissed by other logic.

Reply:
I don't understand what we get from this portion. How does presenting off the rootViewController vs. the frontMostVC make us more robust? We can still be dismissed by any similar logic in the app.

e.g. here how we' "dismiss whatever is presented and present this VC instead", hopefully we're not doing that anywhere else 😬 

Also seems like a minor UX degredation - previously after ending the call, the user would be back where they left off, right? But now we're always returning them to the home view before presenting the call screen.

Does this work if there are multiple VC's presented atop the rootView? seems like we want something like an "unwind" segue.

I'm left feeling like this portion of the changes asks more questions than it answers. I'm OK with working through the questions if you want to, but otherwise I'm inclined to just merge the second half of the PR.

The other changes in this PR LGTM